tests/test_linalg.py

```diff
@@ -0,0 +1,100 @@
+import pykokkos as pk
+from pykokkos.linalg.l3_blas import dgemm
+
+import numpy as np
+from numpy.testing import assert_allclose
+import pytest
+
+
+@pytest.mark.parametrize("shape_a, shape_b", [
+    ([3, 3], [3, 3]),
+    ([11, 7], [7, 5]),
+])
+def test_dgemm_shape(shape_a, shape_b):
+    view_a = pk.View(shape_a, dtype=pk.double)
+    view_b = pk.View(shape_b, dtype=pk.double)
+    alpha = 1.0
+    actual = dgemm(alpha=alpha,
+                   view_a=view_a,
+                   view_b=view_b)
+    # shape requirement of C
+    assert actual.shape == (view_a.shape[0], view_b.shape[1])
+
+
+@pytest.mark.parametrize("alpha, a, b, beta, expected_c", [
+    (1.0,
+     np.array([[1, 7, 9],
+               [3, 1, 3],
+               [5, 5, 22]], dtype=np.float64),
+     np.array([[9, 0, 2],
+               [77, 100, 4],
+               [1, 500, 9]], dtype=np.float64),
+     1.0,
+     np.array([[557.,  5200., 111.],
+               [107.,  1600., 37.],
+               [452., 11500., 228.]], dtype=np.float64),
+     ),
+    (7.7,
+     np.array([[8, 7, 1, 200, 55.3],
+               [99.2, 1.11, 2.02, 17.7, 900.2],
+               [5.01, 15.21, 22.07, 1.09, 22.22],
+               [1, 2, 3, 4, 5]], dtype=np.float64),
+     np.array([[9, 0, 2],
+               [77, 100, 4],
+               [1, 500, 9],
+               [226.68, 11.61, 12.12],
+               [17.7, 200.10, 301.17]], dtype=np.float64),
+     1.0,
+     np.array([[361336.437, 112323.981, 147314.0977],
+               [161130.7082, 1397215.1809, 2090925.5906],
+               [14466.03004, 131014.55213, 53705.17614],
+               [8941.394, 21151.438, 12253.241]], dtype=np.float64),
+     ),
+    # this case just expands b by one column from
+    # case above
+    (7.7,
+     np.array([[8, 7, 1, 200, 55.3],
+               [99.2, 1.11, 2.02, 17.7, 900.2],
+               [5.01, 15.21, 22.07, 1.09, 22.22],
+               [1, 2, 3, 4, 5]], dtype=np.float64),
+     np.array([[9, 0, 2, 19],
+               [77, 100, 4, 19],
+               [1, 500, 9, 19],
+               [226.68, 11.61, 12.12, 19],
+               [17.7, 200.10, 301.17, 20]], dtype=np.float64),
+     1.0,
+     np.array([[361336.437, 112323.981, 147314.0977, 40117.],
+               [161130.7082, 1397215.1809, 2090925.5906, 156191.189],
+               [14466.03004, 131014.55213, 53705.17614, 9768.374],
+               [8941.394, 21151.438, 12253.241, 2233.]], dtype=np.float64),
+     ),
+])
+def test_dgemm_vs_scipy(alpha,
+                        a,
+                        b,
+                        beta,
+                        expected_c):
+    # test against expected results from
+    # scipy.linalg.blas.dgemm
+    # NOTE: it seems that SciPy may ignore beta
+    # in its wrapper perhaps?
+    # see: https://github.com/scipy/scipy/issues/17594
```

Comment:
As clarified over there, I suppose we'll need to add a `view_c` argument and test permutations with `beta` that way. At least, that seems to be how NETLIB LAPACK implementation works.